### DISCUSSION TITLE
Error when reading col names in a loop

### DISCUSSION THREAD

**Discussion by benjaminplong** [ERGO-Code/HiGHS#1706] [Q&A]
I have loaded a model through an LP file and want to get the names of the variables from HiGHs. I am attempting to do this in a for loop, but only make it a few iterations before encountering errors. Some examples of the errors are:

malloc(): invalid size (unsorted)
free(): invalid pointer

I think there was another one about "corrupted size < previous", but I cannot reproduce it.

Has anyone encountered this type of issue?

**Comment by jajhall** [comment]
I've not heard of anything like this before. What HiGHS calls were you making?

**Reply by benjaminplong** [reply]
I am just calling Highs_getColName in a for loop. I am running inside a Java JVM so there might be some weird things going on. I don’t have a C++ repro but if you expect it to work fine there I will reach out to the JNA community to see if they can help.

**Comment by jajhall** [comment]
OK, so the closest I can get to this is in the C API. It's soon done, so I'll convet this to an issue as a reminder

**Reply by benjaminplong** [reply]
Are you confirming issues with this method or are you saying that you will create an issue to investigate?

**Reply by jajhall** [reply]
I've just created the issue so that I remember to look at it. I'm busy with teaching for a few more days...

**Reply by benjaminplong** [reply]
Thank you. I will also see if I can make a C++ reproduction and discuss with the JNA community if I am making the call correctly from Java

**Reply by benjaminplong** [reply]
@jajhall I was able to identify a problem in the Java code. I was using the wrong type to access the "char *" output. You can close your issue. Thank you for responding quickly though.

**Reply by jajhall** [reply]
Thanks
